server/src/services/server.service.ts

```diff
@@ -147,7 +150,14 @@ export class ServerService implements OnEvents {
       serverStats.videos += usage.videos;
       serverStats.usage += usage.usage;
       serverStats.usageByUser.push(usage);
+
+      this.metricRepository.user.addToGauge(`immich.user.photos`, usage.photos, {},{"userName": user.userName})
+      this.metricRepository.user.addToGauge(`immich.user.videos`, usage.videos, {},{"userName": user.userName})
+      this.metricRepository.user.addToGauge(`immich.user.usage`, usage.usage, {},{"userName": user.userName})
     }
+    this.metricRepository.host.addToGauge(`immich.host.photos`, serverStats.photos)
+    this.metricRepository.host.addToGauge(`immich.host.videos`, serverStats.videos)
+    this.metricRepository.host.addToGauge(`immich.host.usage`, serverStats.usage)
```

Comment:
Btw, am I being stupid or you cannot send metrics without creating "sums"

Currently this keeps adding up the one photo I have. I can't use a Counter because it can decrease and the Gauge is adding stuff.

I am very noob in Otel, I am used to send prometheus data with bash or python directly :)

Reply:
It's not as easy to do it for persistent data like that. Options:
 - Increment whenever an asset gets created (might be difficult to keep in sync with actual)
 - Query Postgres for the data (accurate, but counting in Postgres is very slow since it's a full table scan, so doing it all the time is a very bad idea).
 - Have a trigger in Postgres that increments values in a separate counting table when assets are added or removed (fast retrieval for metrics, but makes insertions slower, not sure by how much).

Reply:
Exactly, I'll think about it! It is missing that on my grafana

Reply:
But to just set it in this handler and not automatically, I'm sure you can set the gauges to explicitly be those values. The metric repo doesn't expose everything from the underlying otel gauges.

Reply:
Ah, I think the up/down counter is different from a gauge. [This](https://opentelemetry.io/docs/languages/js/instrumentation/#using-observable-async-gauges) should be what you're looking for.

Reply:
I tried that but then it got stuck on 1... I need to try again. Thank you

Reply:
I also need to find an endpoint that gets called more often (that one is just when the admin opens that panel) or create a cron for that

Reply:
You can add it to the nightly jobs [here](https://github.com/immich-app/immich/blob/d23aa5e8e21bc3049fa65102582105025469a24a/server/src/services/job.service.ts#L205).

Reply:
> You can add it to the nightly jobs [here](https://github.com/immich-app/immich/blob/d23aa5e8e21bc3049fa65102582105025469a24a/server/src/services/job.service.ts#L205).

That might not be needed with the Observable Gauge since it triggers the function on scrape

I attempted with it but the value is still stuck at 0. Do you have any other ideas?

Reply:
It shouldn't make a new observable gauge with each call. The gauge should be called when the service is created and reference variables in the service. These variables should be updated when the user statistics handler is called.

*Definitely don't let this run every time the endpoint is scraped*.